std/algorithm/comparison.d

```diff
@@ -604,85 +604,85 @@ Returns:
 
 */
 int cmp(alias pred = "a < b", R1, R2)(R1 r1, R2 r2)
-if (isInputRange!R1 && isInputRange!R2 && !(isSomeString!R1 && isSomeString!R2))
+if (isInputRange!R1 && isInputRange!R2 || isSomeString!R1 && isSomeString!R2)
 {
-    for (;; r1.popFront(), r2.popFront())
+    static if (isInputRange!R1 && isInputRange!R2)
```

Comment:
this will be true for all inputs, the is some string overload needs to come first